# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/MediaConduitInterface.h
# Commit: 7057437ddf19
# Full Hash: 7057437ddf194b62a158ce84345802e8b2bc6ca8
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 6
# Description:
#   Bug 1654112 - Constify VideoConduit members that are based on prefs. r=ng
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D112848
# ==============================================================================

diff -r 01569f2b1141 -r 7057437ddf19 dom/media/webrtc/libwebrtcglue/MediaConduitInterface.h
--- a/dom/media/webrtc/libwebrtcglue/MediaConduitInterface.h	Tue Apr 06 18:29:30 2021 +0200
+++ b/dom/media/webrtc/libwebrtcglue/MediaConduitInterface.h	Wed Apr 07 09:44:33 2021 +0200
@@ -6,18 +6,14 @@
 #define MEDIA_CONDUIT_ABSTRACTION_
 
 #include <vector>
-#include <set>
 #include <functional>
 #include <map>
 
 #include "CodecConfig.h"
 #include "ImageContainer.h"
-#include "jsapi/PeerConnectionCtx.h"
 #include "jsapi/RTCStatsReport.h"
 #include "MediaConduitErrors.h"
 #include "mozilla/media/MediaUtils.h"
-#include "mozilla/RefCounted.h"
-#include "TaskQueueWrapper.h"
 #include "VideoTypes.h"
 #include "WebrtcVideoCodecFactory.h"
 #include "RtcpEventObserver.h"
@@ -25,7 +21,6 @@
 #include "mozilla/dom/RTCRtpSourcesBinding.h"
 
 // libwebrtc includes
-#include "api/video/builtin_video_bitrate_allocator_factory.h"
 #include "api/video/video_frame_buffer.h"
 #include "call/call.h"
 
@@ -319,6 +314,19 @@
  */
 class VideoSessionConduit : public MediaSessionConduit {
  public:
+  struct Options {
+    bool mVideoLatencyTestEnable = false;
+    // All in bps.
+    int mMinBitrate = 0;
+    int mStartBitrate = 0;
+    int mPrefMaxBitrate = 0;
+    int mMinBitrateEstimate = 0;
+    bool mDenoising = false;
+    bool mLockScaling = false;
+    uint8_t mSpatialLayers = 1;
+    uint8_t mTemporalLayers = 1;
+  };
+
   /**
    * Factory function to create and initialize a Video Conduit Session
    * @param  webrtc::Call instance shared by paired audio and video
@@ -328,7 +336,8 @@
    */
   static RefPtr<VideoSessionConduit> Create(
       RefPtr<WebrtcCallWrapper> aCall,
-      nsCOMPtr<nsISerialEventTarget> aStsThread, std::string aPCHandle);
+      nsCOMPtr<nsISerialEventTarget> aStsThread, Options aOptions,
+      std::string aPCHandle);
 
   enum FrameRequestType {
     FrameRequestNone,